mlkem/ntt.c

```diff
@@ -95,37 +117,54 @@ void poly_ntt(poly *p) { ntt_native(p); }
  * Description: Computes inverse of negacyclic number-theoretic transform (NTT)
  *              of a polynomial in place;
  *              inputs assumed to be in bitreversed order, output in normal
- *order
+ *              order
+ *
+ *              The input is assumed to be in bitreversed order.
+ *
+ *              The output polynomail is in normal order, and
+ *              coefficient-wise bound by INVNTT_BOUND in absolute value.
+ *
+ *              Values of INVNTT_BOUND:
+ *              - C: 3/4 q
+ *              - AArch64: 8*q
  *
  * Arguments:   - uint16_t *a: pointer to in/output polynomial
  **************************************************/
 #if !defined(MLKEM_USE_NATIVE_INTT)
 // REF-CHANGE: Removed indirection poly_invntt_tomont -> invntt()
+// REF-CHANGE: Moved scalar multiplication with f ahead of the core invNTT
 void poly_invntt_tomont(poly *p) {
   unsigned int start, len, j, k;
   int16_t t, zeta;
   const int16_t f = 1441;  // mont^2/128
   int16_t *r = p->coeffs;
 
+  for (j = 0; j < 256; j++) {
```

Comment:
Can you justify why this still returns the right answer?

Reply:
The modular arithmetic is unchanged, but we're moving the scaling step to the front so it acts as a reduction of the input data. Does that answer your question?

Reply:
Yes... makes sense, but it's such an obvious departure from the reference implementation that an explanatory comment with worth-while.

Reply:
Oh, that's a very good point. We _do_ want those `REF-CHANGE` comments, at least in major deviations like this. I'll add it.

Reply:
Oh, actually there is already a comment at the top of the function

Reply:
Noted